sdk/python/tests/nem/test_TransactionFactory.py

```diff
@@ -38,19 +38,39 @@ def create_factory(self, type_parsing_rules=None):
 	def create_transaction(self, factory):
 		return factory.create
 
+	# region rules
+
+	def test_rules_contain_expected_hints(self):
+		# Act:
+		factory = factory = self.create_factory()
+
+		# Assert:
+		expected_rule_names = [
+			'Amount', 'Height', 'Timestamp',
+
+			'BlockType', 'LinkAction', 'MessageType', 'MosaicSupplyChangeAction', 'MosaicTransferFeeType',
+			'MultisigAccountModificationType', 'NetworkType', 'TransactionType',
+
+			'struct:Message', 'struct:NamespaceId', 'struct:MosaicId', 'struct:Mosaic', 'struct:SizePrefixedMosaic', 'struct:MosaicLevy',
+			'struct:MosaicProperty', 'struct:SizePrefixedMosaicProperty', 'struct:MosaicDefinition',
+			'struct:MultisigAccountModification', 'struct:SizePrefixedMultisigAccountModification',
+
+			'Address', 'Hash256', 'PublicKey',
+
+			'array[SizePrefixedMosaic]', 'array[SizePrefixedMosaicProperty]', 'array[SizePrefixedMultisigAccountModification]'
+		]
+		self.assertEqual(set(expected_rule_names), set(factory.factory.rules.keys()))
+
+	# endregion
+
 	# region create
 
 	def test_can_create_known_transaction_with_multiple_overrides(self):
 		# Arrange:
-		#  * hint on `rental_fee_recipient` is pod:Address,
-		#   type parsing rule that is getting created is for STRING 'Address'
-		# * same happens for `rental_fee`
-		#  * signer_public_key is handled differently in Transaction factory, via hint:
-		#    {'signer_public_key': PublicKey} which maps it to SDK type CryptoTypes.PublicKey
 		factory = self.create_factory({
-			'Address': lambda x: x + ' but amazing',
-			'Amount': lambda _: 654321,
-			PublicKey: lambda address: address + ' PUBLICKEY'
+			Address: lambda x: f'{x} but amazing',
+			Amount: lambda _: 654321,
```

Comment:
this now tests slightly different thing, but not sure if that is a problem

Reply:
yea, but, i think it is ok